fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java

```diff
@@ -277,26 +277,34 @@ private static void checkCommandLineOptions(CommandLineOptions cmdLineOpts) {
         // go on
     }
 
-    private static boolean createAndLockPidFile(String pidFilePath) throws IOException {
+    private static boolean createAndLockPidFile(String pidFilePath) {
         File pid = new File(pidFilePath);
-        try (RandomAccessFile file = new RandomAccessFile(pid, "rws")) {
-            FileLock lock = file.getChannel().tryLock();
-            if (lock == null) {
-                return false;
-            }
+        for (int i = 0; i < 3; i++) {
+            try (RandomAccessFile file = new RandomAccessFile(pid, "rws")) {
+                FileLock lock = file.getChannel().tryLock();
+                if (lock == null) {
+                    LOG.warn("get pid file lock failed, retried: {}", i);
+                    Thread.sleep(10000L);
+                    continue;
```

Comment:
why not throw a exception

Reply:
The caller determines whether the lock acquisition was successful based on the return of true/false.

Reply:
https://github.com/StarRocks/starrocks/pull/5615/files#diff-4e26971c00c7e12e700dcfb24c434a2c711047605ab45e1407c3a271a13b5c95L81

Reply:
I want to say is that you throw an exception here, the following catch can handle this exception.

Reply:
done